### PR TITLE
Add firewall support safer-cluster modules

### LINKED ISSUE #569: configure additional firewall rules for a safer-cluster(-update-variant)
It is currently not possible to enable `add_cluster_firewall_rules` and configure additional firewall rules for a safer-cluster(-update-variant). Is this by design?

### PR DESCRIPTION
**Pull request by Dev25**
Closes #569

### CODE REVIEW

File: autogen/main/firewall.tf.tmpl [hunk 1]
```diff
@@ -48,9 +48,11 @@ resource "google_compute_firewall" "intra_egress" {
   allow { protocol = "esp" }
   allow { protocol = "ah" }
 
+  {% if not private_cluster %}
   depends_on = [
     google_container_cluster.primary,
   ]
+  {% endif %}
```
Comment (Dev25) [inline autogen/main/firewall.tf.tmpl:55]: Private clusters have a pre defined master cidr so no need to wait for cluster creation to determine master ip range.

**Comment by Dev25** [comment]
@bharathkkb Never added a new inspec test before, I've added a firewall existence check based on another tf module [test](https://github.com/terraform-google-modules/terraform-google-network/blob/master/test/integration/submodule_firewall/controls/gcp.rb)

**Comment by bharathkkb** [comment]
@Dev25 I think we need a new verifier system block in `.kitchen.yml` for using inspec.
```yaml
  - name: "safer_cluster"
    driver:
      root_module_directory: test/fixtures/safer_cluster
    verifier:
      systems:
        - name: safer_cluster
          backend: local
        - name: inspec-gcp
          backend: gcp
          controls:
            - network
```

**Comment by Dev25** [comment]
Right thanks! Hopefully that does the trick.

**Comment by bharathkkb** [comment]
@Dev25 I just took a look at another tests we have in this module and it looks like we need another small tweak, to explicitly say only the gcloud control should use local backend. PFA new version.
```yaml
  - name: "safer_cluster"
    driver:
      root_module_directory: test/fixtures/safer_cluster
    verifier:
      systems:
        - name: safer_cluster
          backend: local
          controls:
            - gcloud
        - name: inspec-gcp
          backend: gcp
          controls:
            - network
```

File: test/integration/safer_cluster/controls/network.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+project_id   = attribute('project_id')
+cluster_name = attribute('cluster_name')
+
+control "network" do
+  title "gcp network configuration"
+  describe google_compute_firewalls(project: project_id) do
+    its('firewall_names') { should include "gke-#{cluster_name[0..25]}-intra-cluster-egress" }
```
Comment (bharathkkb) [inline test/integration/safer_cluster/controls/network.rb:21]: Not a ruby expert but looking at this [1] wont we need to discard the last char as we need length to be at most 25chars? So `[0...25]` or `[0,25]`
[1]: https://stackoverflow.com/questions/9690801/difference-between-double-dot-and-triple-dot-in-range-generation

Comment (bharathkkb) [reply test/integration/safer_cluster/controls/network.rb:21]: @Dev25  Saw this error in CI
Cluster name: `safer-cluster-cluster-7yid`
Actual FW name when I checked: `gke-safer-cluster-cluster-7yi-intra-cluster-egress`
What we checked for in the test: `gke-safer-cluster-cluster-7yid-intra-cluster-egress`
```
  ×  network: gcp network configuration (2 failed)
     ×  google_compute_firewalls firewall_names should include "gke-safer-cluster-cluster-7yid-intra-cluster-egress"
     expected ["default-allow-icmp", "default-allow-internal", "default-allow-rdp", "default-allow-ssh", "gke-deplo...node-http-hc", "k8s-fw-a2ee97657700f434d9340944b00a5b61", "k8s-fw-ae0f40471c88f42199bb5196a6394f69"] to include "gke-safer-cluster-cluster-7yid-intra-cluster-egress"
     ×  google_compute_firewalls firewall_names should include "gke-safer-cluster-cluster-7yid-webhooks"
     expected ["default-allow-icmp", "default-allow-internal", "default-allow-rdp", "default-allow-ssh", "gke-deplo...node-http-hc", "k8s-fw-a2ee97657700f434d9340944b00a5b61", "k8s-fw-ae0f40471c88f42199bb5196a6394f69"] to include "gke-safer-cluster-cluster-7yid-webhooks"
```

Comment (Dev25) [reply test/integration/safer_cluster/controls/network.rb]: Changed to `[0,25]`, thanks for helping debug this.

**Pull request merged**